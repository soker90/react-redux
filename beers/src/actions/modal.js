import modalAction from '../constants/actions/modal'

export default {
  openModal: modalData => ({type: modalAction.OPEN_MODAL, modalData}),
  closeModal: () => ({type: modalAction.CLOSE_MODAL})
}
