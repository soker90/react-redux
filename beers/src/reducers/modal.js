import modalAction from '../constants/actions/modal'

export function initialState () {
  return {
    open: false,
    title: '',
    body: '',
    callback: null,
    extraParams: {}
  }
}

function openModal(state, {modalData}) {
  return {
    ...state,
    open: true,
    title: modalData.title,
    body: modalData.body,
    callback: modalData.callback ? modalData.callback : null,
    extraParams: modalData.extraParams ? modalData.extraParams : {}
  }
}

function closeModal(state) {
  return {
    ...state,
    open: false,
    title: '',
    body: '',
    callback: null,
    extraParams: {}

  }
}

export default function (state = initialState(), action) {
  switch (action.type) {
    case modalAction.OPEN_MODAL:
      return openModal(state, action)
    case modalAction.CLOSE_MODAL:
      return closeModal(state, action)
    default:
      return state
  }
}
