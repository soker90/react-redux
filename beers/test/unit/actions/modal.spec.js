// modal.spec.js
import modal from '../../../src/actions/modal'

describe('modal actions', () => {
  test('openModal', () => {
    expect(modal.openModal()).toMatchSnapshot()
  })
  test('closeModal', () => {
    expect(modal.closeModal()).toMatchSnapshot()
  })
})
