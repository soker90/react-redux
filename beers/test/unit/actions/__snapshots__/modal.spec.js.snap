exports[`modal actions closeModal 1`] = `
Object {
  "type": "CLOSE_MODAL",
}
`;

exports[`modal actions openModal 1`] = `
Object {
  "modalData": undefined,
  "type": "OPEN_MODAL",
}
`;
